src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,386 @@
+            return fieldInfo
+        }
+        setDirectoryData(constructDirectoryObject(fieldsWithDefaults, records));
+        
+        
+    }, [records, fieldsWithDefaults])
+
+    // useEffect(() => {
+    //     const propRecords = [];
+    //     for(let i = 0; i <= directoryData.length; i++){ 
+    //         records.push(getRecord(i))
+            
+    //     }
+    //     setRecordList(records)
+    // }, [directoryData])
+
+    // const getRecord = (index) => Object.values(directoryData).map((field) => {
+    //     return field["values"][index]
+    // })
+    
+    const viewHeader = () => Object.values(directoryData).map((field) => {
+
+        return (
+            <Grid item xs={field["col-width"]} md={field["col-width"]} >
+                <Typography align={field.align} style={{color: 'white', fontWeight: '500'}}>
+                    {field.label}
+                </Typography>
+            </Grid>
+        )
+    })
+
+    
+    const viewRecordRow = (record, index) => {
+        const recordElements = [];
+
+        Object.values(directoryData).forEach((field, index) => {
+            recordElements.push(
+                <Grid item xs={field["col-width"]} md={field["col-width"]}>
+                    <Typography align={field["align"]}>
+                        {Object.values(record)[index]}
+                    </Typography>
+                </Grid>
+            );
+        })
+
+        return (
+            <Paper square={true} className={"category-row"} >
+                <Grid container alignItems={"center"}>
+                
+                    {recordElements}
+                    <Grid item xs md>
+                    <IconButton
+                        onClick={editRecord(index)}
+                        >
+                    <EditIcon/>
+                    </IconButton>
+                </Grid>
+                </Grid>
+
+            </Paper>
+        )
+    };
+
+    // const editRecord = (id) => (e) => {
+
+    //     e.preventDefault();
+
+    //     let editingRecord = recordList.find((record) => {return record.id === id});
+    //     let recordToUpload = {};
+
+    //     if(editingRecord) {
+    //         Object.keys(editingRecord).forEach((key) => {
+    //             recordToUpload[key] = editingRecord[key]
+    //         });
+    //         props.updateFunction(id, recordToUpload);
+    //     }
+    //     editingRecord.editing = !editingRecord.editing;
+    //     let updatedRecordList = recordList.map((record) => {
+    //         if(record.id == id){
+    //             return editingRecord;
+    //         } else {
+    //             return record;
+    //         }
+    //     });
+    //     setDirectoryData(updatedRecordList);
+    // };
+
+    const editRecord = (index) => (e) => {
+        e.preventDefault();
+        const recordToEdit = {};
+
+        let mutableDirectoryData = directoryData;
+        Object.keys(directoryData).map(key => {
+            if (key == "editing") {
```

Comment:
Why do you need this case? If you don't need it then just check the other conditions:
```
key == "ids" ? recordToEdit.id = directoryData.ids.index : recordToEdit.key = directoryData.key.values.index
```